src/context/auth-context.js

```diff
@@ -0,0 +1,51 @@
+import axios from "axios";
+import { useNavigate } from "react-router-dom";
+import { createContext, useContext, useReducer } from "react";
+import { authInitialState, authReducer } from "../reducer/reducer";
+
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const [authState, authDispatch] = useReducer(authReducer, authInitialState);
+  const navigate = useNavigate();
+
+  const loginHandler = async (email, password) => {
+    try {
+      const { data } = await axios.post("/api/auth/login", {
+        email: email,
```

Comment:
Just a suggestion for future reference;

if key and value both are same; you can use only one 
like replace ```email: email, password: password``` with ```email, password```

Reply:
ya sure. will do